### PR TITLE
feat: add paint timing mark to page-load transaction

### LINKED ISSUE #13: Add paint timing marks to the navigation transaction
+ In addition to the current marks that we have like TTFB,  Dom Interactive. It would be nice to also add paint metrics like first-paint and first-contentful-paint. 
+ Its only available in chrome at the moment, We can right now use FCP in our marks since FCP provides better information than FP at the moment.

+ I did some research on this topic before and we could use this as  a reference for adding more paint related metrics for images in future https://github.com/vigneshshanmugam/image-paint

### PR DESCRIPTION
**Pull request by vigneshshanmugam**
+ fixes elastic/apm-agent-js-core#13

### CODE REVIEW

File: src/performance-monitoring/transaction.js [hunk 1]
```diff
@@ -65,11 +65,13 @@ class Transaction {
...
   addNavigationTimingMarks () {
     var marks = utils.getNavigationTimingMarks()
+    var paintMarks = utils.getPaintTimingMarks()
     if (marks) {
       var agent = {
         timeToFirstByte: marks.responseStart,
         domInteractive: marks.domInteractive,
-        domComplete: marks.domComplete
+        domComplete: marks.domComplete,
+        firstContentfulPaint: paintMarks['first-contentful-paint']
```
Comment (vigneshshanmugam) [inline src/performance-monitoring/transaction.js]: @jahtalab how is undefined values treated in UI? Should we discard them from sending to apm-server?

File: src/common/utils.js [hunk 2]
```diff
@@ -173,18 +173,32 @@ var navigationTimingKeys = [
...
+ * Paint Timing Metrics that is available during page load
+ * https://www.w3.org/TR/paint-timing/
+ */
+function getPaintTimingMarks () {
+  var paints = {}
+  if (window.performance.getEntriesByType) {
+    var entries = window.performance.getEntriesByType('paint')
+    for (var i = 0; i < entries.length; i++) {
+      var data = entries[i]
+      paints[data.name] = data.startTime
```
[the remark this answers is not in the document]
Comment (vigneshshanmugam) [reply src/common/utils.js]: No, they are not tied to navigation at all. It should happen whenever the document gets painted and in SPA, if we start capturing navigations then it should also contain the FCP at that point.

[the remark this answers is not in the document]
Comment (jahtalab) [reply src/common/utils.js]: According to https://w3c.github.io/paint-timing/#first-contentful-paint, the first contentful paint contains a `DOMHighResTimeStamp` which has a [time origin](https://www.w3.org/TR/hr-time-2/#dfn-time-origin) which is similar to the `navigationStart` timestamp. However for the `page-load` transactions we calculate all the timings based on `fetchStart` and not the `navigationStart` and these two could be different in certain situations.

[the remark this answers is not in the document]
Comment (vigneshshanmugam) [reply src/common/utils.js]: As discussed, did the diff of fetchStart - navStart

**Pull request merged**